src/wp-includes/http.php

```diff
@@ -710,31 +710,60 @@ function ms_allowed_http_request_hosts( $is_external, $host ) {
  *               PHP_URL_PORT - integer when it does. See parse_url()'s return values.
  */
 function wp_parse_url( $url, $component = -1 ) {
-	$to_unset = array();
-	$url      = (string) $url;
 
-	if ( str_starts_with( $url, '//' ) ) {
-		$to_unset[] = 'scheme';
-		$url        = 'placeholder:' . $url;
-	} elseif ( str_starts_with( $url, '/' ) ) {
-		$to_unset[] = 'scheme';
-		$to_unset[] = 'host';
-		$url        = 'placeholder://placeholder' . $url;
-	}
+	switch ( $component ) {
 
-	$parts = parse_url( $url );
+		case -1:
+			return parse_url( (string) $url );
```

Comment:
You use the typecast `(string)` here and again at the bottom.

I'd either normalize `$url` before the `switch` or simply set `$component = null;` and use the single return at the end to reduce redundancy here.